src/mlpack/core/util/param_checks_impl.hpp

```diff
@@ -299,6 +299,29 @@ inline void ReportIgnoredParam(const std::string& paramName,
   }
 }
 
+inline void CheckInputShape(const size_t x_rows,
+                        const size_t x_cols, 
+                        const size_t y_cols, 
+                        const size_t xt_rows)
+{
+  if(x_rows==0)
+  {
+    Log::Fatal <<"No training data available";
+    return;
+  }
+  if(x_cols!=y_cols)
+  {
+    Log::Fatal << "Number of data points in training data and labels don't match. Number of data points in feature matrix are " << x_cols 
```

Comment:
If this utility is meant to be generic, the datasets may not always make sense to refer to as `training data` and `labels`.